airflow/ti_deps/deps/pool_slots_available_dep.py

```diff
@@ -63,7 +63,7 @@ def _get_dep_statuses(self, ti, session, dep_context=None):
         if ti.state in STATES_TO_COUNT_AS_RUNNING:
             open_slots += ti.pool_slots
 
-        if open_slots <= (ti.pool_slots - 1):
+        if open_slots < ti.pool_slots:
```

Comment:
Could you say a little more about this change?

Reply:
My understanding is: open_slots means how many slots left in the pool (ie available to be consumed/tasks run in), pool_slots means how many slots this specific taskinstance aims to consume. Previously it assumed each task would take at least 1 slot, now I'm making it handle case where task can take 0 slots